File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/1028/BusinessApplication/BA.Web.10.0/Services/AuthenticationService.vb
Imports System.Security.Authentication
Imports System.ServiceModel.DomainServices.Hosting
Imports System.ServiceModel.DomainServices.Server
Imports System.ServiceModel.DomainServices.Server.ApplicationServices
Imports System.Threading
Namespace Web

    '' TODO: 部署此應用程式時切換到安全端點。
    ''       使用者的名稱和密碼只可以使用 https 傳遞。
    ''       若要執行此動作，請將 EnableClientAccessAttribute 上的 RequiresSecureEndpoint 屬性設定為 true。
    ''   
    ''       <EnableClientAccess(RequiresSecureEndpoint:=True)>
    ''
    ''       可以在 MSDN 上取得搭配  https 使用網域服務的詳細資訊。

    ''' <summary>
    ''' 網域服務負責在使用者登入應用程式時對使用者進行驗證。
    '''
    ''' 大部分的功能已由 AuthenticationBase 類別提供。
    ''' </summary>
    <EnableClientAccess()> _
    Public Class AuthenticationService
        Inherits AuthenticationBase(Of User)
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/2052/BusinessApplication/BA.Web/Services/AuthenticationService.cs
namespace $safeprojectname$
{
    using System.Security.Authentication;
    using System.ServiceModel.DomainServices.Hosting;
    using System.ServiceModel.DomainServices.Server;
    using System.ServiceModel.DomainServices.Server.ApplicationServices;
    using System.Threading;

    // TODO: 在部署应用程序时切换为安全终结点。
    //       只应使用 https 传递用户名和密码。
    //       为此，请将 EnableClientAccessAttribute 的 RequiresSecureEndpoint 属性设置为 true。
    //   
    //       [EnableClientAccess(RequiresSecureEndpoint = true)]
    //
    //       有关对域服务使用 https 的更多信息可在 MSDN 上找到。

    /// <summary>
    /// 负责在用户登录应用程序时对用户进行身份验证的域服务。
    ///
    /// AuthenticationBase 类已提供了大部分功能。
    /// </summary>
    [EnableClientAccess]
    public class AuthenticationService : AuthenticationBase<User> { }
}
